File: src/main/scala/mffs/render/button/GuiButtonPressTransferMode.java
package mffs.render.button;

import mffs.base.GuiMFFS;
import mffs.production.TileFortronCapacitor;
import net.minecraft.client.Minecraft;
import universalelectricity.api.vector.Vector2;

public class GuiButtonPressTransferMode extends GuiButtonPress
{
	private TileFortronCapacitor tileEntity;

	public GuiButtonPressTransferMode(int id, int x, int y, GuiMFFS mainGui, TileFortronCapacitor tileEntity)
	{
		super(id, x, y, new Vector2(), mainGui);
		this.tileEntity = tileEntity;
	}

	@Override
	public void drawButton(Minecraft minecraft, int x, int y)
	{
		String transferName = this.tileEntity.getTransferMode().name().toLowerCase();
		char first = Character.toUpperCase(transferName.charAt(0));
		transferName = first + transferName.substring(1);
		this.displayString = "transferMode" + transferName;
		this.offset.y = 18 * this.tileEntity.getTransferMode().ordinal();

		super.drawButton(minecraft, x, y);
	}
}

File: src/main/scala/mffs/render/button/GuiButtonPressTransferMode.scala
package mffs.render.button

import mffs.base.GuiMFFS
import mffs.production.TileFortronCapacitor
import net.minecraft.client.Minecraft
import universalelectricity.core.transform.vector.Vector2

class GuiButtonPressTransferMode(id: Int, x: Int, y: Int, mainGui: GuiMFFS, tile: TileFortronCapacitor) extends GuiButtonPress(id, x, y, new Vector2, mainGui)
{
  override def drawButton(minecraft: Minecraft, x: Int, y: Int)
  {
    var transferName = tile.getTransferMode.toString
    val first = Character.toUpperCase(transferName.charAt(0))
    transferName = first + transferName.substring(1)
    this.displayString = "transferMode" + transferName
    val offsetY = 18 * tile.getTransferMode.id
    super.drawButton(minecraft, x, y)
  }
}